extension/extensionlimiter/extensionlimiter.go

```diff
@@ -0,0 +1,60 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package extensionlimiter // import "go.opentelemetry.io/collector/extension/extensionlimiter"
+
+import (
+	"context"
+)
+
+// Option is passed to limiter providers.
+//
+// NOTE: For data-specific or tenant-specific limits we will extend
+// providers with Options and add a Config type, but none are
+// supported yet and this PR contains only interfaces, not need for
+// options in core repository components.
+type Option interface {
+	apply()
+}
+
+// Checker is for checking when a limit is saturated.  This can be
+// called prior to the start of work to check for limiter saturation.
+type Checker interface {
```

Comment:
Do we need to support a limiter that is just "Checker" like the current memory limiter? I would suggest yes.

Reply:
Yes. So, would you say there are three fundamental limiter APIs, the Rate/Resource and Basic? There is a provider for each -- does it make sense, to you, that there Rate/Resource providers embed the basic limiter provider (as I have called `CheckerProvider` in this PR)?

Callers would be expected to check for more-specific extension APIs before the less-specific one.

Reply:
I like it